apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -1192,7 +1192,7 @@ func (m *ModelManagerAPI) getModelInfo(ctx context.Context, tag names.ModelTag,
 		}
 	}
 	if canSeeMachinesAndSecrets {
-		if info.Machines, err = common.ModelMachineInfo(ctx, st, m.machineService); shouldErr(err) {
+		if info.Machines, err = common.ModelMachineInfo(ctx, st, modelDomainServices.Machine()); shouldErr(err) {
```

Comment:
This is the heart of the fix. This facade is tricky because it needs access to the controller model and the current model, so we need to pick the domain services accordingly.